### PR TITLE
Feature/20 add picture to dataset

### PR DESCRIPTION
**Pull request by wattsecs**
Features:
- Take a picture
- View the picture that was taken
- Retake a picture
- Select the category
- Save the picture
- Share the result with another activity using the provided contract

Todo:
- clean up commit history
- can't get testing permission with denial working

### CODE REVIEW

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/AddPictureActivity.kt [hunk 1]
```diff
@@ -61,34 +68,80 @@ class AddPictureActivity : AppCompatActivity() {
...
+            putParcelable("image", capturedImageUri)
+        }
+        returnIntent.putExtra("result", bundle)
+        setResult(Activity.RESULT_OK, returnIntent)
+        finish()
+    }
+
+    @Suppress("UNUSED_PARAMETER")
+    private fun onTakePicture(view: View) {
+        val executor = Executors.newSingleThreadExecutor()
```
Comment (bbjubjub2494) [inline app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/AddPictureActivity.kt:104]: What's the reason we need a brand new thread for this? Seems like all it does is `Log.e`

**Review by Nitwix — APPROVED** [review]
Seems very good! I didn't have time to read every line in detail but in general it seems good.

**Review by bbjubjub2494 — APPROVED** [review]
Good work, kudos on the tenacity.

File: app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/AddPictureActivity.kt [hunk 2]
```diff
@@ -61,34 +68,80 @@ class AddPictureActivity : AppCompatActivity() {
...
 
-    private fun cameraIsAvailable(): Boolean {
-        return applicationContext.packageManager.hasSystemFeature(PackageManager.FEATURE_CAMERA_ANY)
-    }
+    /**
+    The ActivityResultContract which should be used when launching this activity.
+    The launch argument is an Array<String> containing the categories to select from.
+    The return value is a Pair containing the selected category as a first element and the Uri pointing to the image as a second element
+     */
+    object AddPictureContract : ActivityResultContract<Array<String>, Pair<String, Uri>?>() {
```
[the remark this answers is not in the document]
Comment (wattsecs) [reply app/src/main/java/com/github/HumanLearning2021/HumanLearningApp/AddPictureActivity.kt:76]: Yes that would be better indeed. I wanted Category to be finalized before adapting it.

**Pull request merged**